addons/website_event_meet/controllers/main.py

```diff
@@ -0,0 +1,104 @@
+            "current_lang": lang,
+            "available_languages": event.sudo().meeting_room_ids.mapped("lang_id"),
+            "open_room": int(open_room) if open_room else None,
+            "is_event_manager": request.env.user.has_group("event.group_event_manager"),
+        }
+
+        return request.render("website_event_meet.template_meeting_rooms", values)
+
+    @http.route(["/event/create_meeting_room"], type="http", auth="public", methods=["POST"])
+    def create_meeting_room(self):
+        name = http.request.params.get("name")
+        summary = http.request.params.get("summary")
+        target_audience = http.request.params.get("audience")
+        lang = http.request.params.get("lang")
+        max_capacity = http.request.params.get("capacity")
+        event_id = int(http.request.params.get("event"))
+
+        # get the record to be sure they really exist
+        event = request.env["event.event"].browse(event_id).exists()
+        lang = request.env["res.lang"].search([("code", "=", lang)], limit=1)
+
+        if not event or not event.website_published or not lang:
+            raise Forbidden()
+
+        _logger.info("New meeting room (%s) create by %s" % (name, request.httprequest.remote_addr))
+
+        meeting_room = request.env["event.meeting_room"].sudo().create(
+            {
+                "name": name,
+                "summary": summary,
+                "target_audience": target_audience,
+                "max_capacity": max_capacity,
+                "lang_id": lang.id,
+                "pinned": False,
+                "event_id": event.id,
+            },
+        )
+
+        return redirect("/event/%s/meeting_rooms?open_room=%i" % (slug(event), meeting_room.id))
+
+    @http.route(["/event/update_participant_count"], type="json", auth="public")
+    def update_participant_count(self, meeting_room_id, joined, participant_count):
+        meeting_room = request.env["event.meeting_room"].browse(meeting_room_id).exists()
+        if not meeting_room or not meeting_room.sudo().event_id.website_published:
+            raise Forbidden()
+
+        request.env.cr.execute(
+            """
+            WITH req AS (
+                SELECT id
+                  FROM event_meeting_room
+                 WHERE id = %s
+                   FOR UPDATE SKIP LOCKED
+            )
+            UPDATE event_meeting_room AS mr
+               SET participant_count = %s,
+                   last_joined = CASE WHEN %s THEN NOW() ELSE last_joined END,
+                   last_activity = NOW(),
+                   max_participant_reached = GREATEST(max_participant_reached, %s)
+              FROM req
+             WHERE mr.id = req.id;
+            """,
+            [meeting_room_id, participant_count, joined, participant_count]
+        )
+
+    @http.route(["/event/active_langs"], type="json", auth="public")
+    def active_langs(self):
+        return request.env["res.lang"].sudo().search_read([], ["name", "code"])
```

Comment:
don't we have already a route to get the active lang ? (maybe in website directly ?)

Reply:
Indeed :) (`/website/get_languages`)

Reply:
@dbeguin Hum, my bad, this route is for the langs of the website, not the active lang on the database

`/web/session/get_lang_list` return all langs, active or not, so I can't use it

Si I don't think there's already a route for that in the "upper module" from which I inherit